src/signed-xml.ts

```diff
@@ -339,13 +340,14 @@ export class SignedXml {
   private calculateSignatureValue(doc: Document, callback?: ErrorFirstCallback<string>) {
     const signedInfoCanon = this.getCanonSignedInfoXml(doc);
     const signer = this.findSignatureAlgorithm(this.signatureAlgorithm);
-    if (this.privateKey == null) {
+    const privateKey = this.getPrivateKey();
+    if (privateKey === undefined) {
```

Comment:
Why is `null` now permissible?

Reply:
`getPrivateKey()` will never return null.
It could never have been null to begin with. It could only be undefined.

Reply:
That doesn't appear to be true. While the typing system would prevent anything other than those values, not everyone uses Typescript. I see no code that would prevent any value from being used for `privateKey`.

Reply:
I dont think thats a rabbit hole worth going down.
By that logic `this.privateKey == null` also does nothing, because i could set privateKey=1, privateKey='' etc.., and that check would not catch it.
You cant reasonably expect the library to work if you set privateKey to null.
In any case an error will be thrown by the crypto library.